src/services/ExperianService.ts

```diff
@@ -0,0 +1,124 @@
+		const token = await this.checkExperianToken();
+		if (token) {
+			return token;
+		} else {		
+			this.logger.info("requestExperianToken - no valid token - trying to generate new Experian token");
+			try {
+				const params = {
+					username: clientUsername,
+					password: clientPassword,
+					client_id: clientId,
+					client_secret: clientSecret,
+				};
+
+				const config: AxiosRequestConfig<any> = {
+    				headers: {
+    					"Content-Type": "application/json",
+						"X-Correlation-Id": randomUUID(),
+						"X-User-Domain": "cabinetofficegds.com",
+    				},
+				};
+
+				const { data }: { data: ExperianTokenResponse } = await axios.post(
+					`${this.experianBaseUrl}${Constants.EXPERIAN_TOKEN_ENDPOINT_PATH}`,
+					params,
+					config,
+				);
+				this.logger.info("Received response from experian token endpoint");
+				await this.saveExperianToken(data);
+				return data;
+			} catch (error: any) {
+				this.logger.error({ message: "Error generating experian token", statusCode: error?.response?.status, messageCode: MessageCodes.FAILED_GENERATING_EXPERIAN_TOKEN });
+				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error generating experian token");
+			}
+		}
+	}
+
+	async saveExperianToken(token : ExperianTokenResponse): Promise<void> {
+		this.logger.info({ message: `Updating ${this.experianTokenTableName} with new Experian token` });
+
+		const putExperianTokenCommand = new PutCommand({
+			TableName: this.experianTokenTableName,
+			Item: { 
+				id: "1",
+				issued_at: token.issued_at,
+				expires_in: token.expires_in,
+				token_type: token.token_type,
+				access_token: token.access_token,
+				refresh_token: token.refresh_token,
+			},
+		});
+
+		try {
+			await this.dynamo.send(putExperianTokenCommand);
+			this.logger.info({ message: "Stored new Experian token" });
+		} catch (error) {
+			this.logger.error({ message: "Error storing new Experian token", messageCode: MessageCodes.FAILED_GENERATING_EXPERIAN_TOKEN, error });
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error storing new Experian token");
+		}
+	}
+
+	async checkExperianToken(): Promise<ExperianTokenResponse | undefined> {
+		this.logger.debug("Fetching Experian token from table " + this.experianTokenTableName);
+		const getExperianTokenCommand = new GetCommand({
+			TableName: this.experianTokenTableName,
+			Key: {
+				id: "1",
+			},
+		});
+
+		let token;
+		try {
+			token = await this.dynamo.send(getExperianTokenCommand);
+		} catch (error) {
+			this.logger.error({ message: "getExperianTokenById - failed executing get from dynamodb:" }, {
+				messageCode: MessageCodes.FAILED_FETCHING_EXPERIAN_TOKEN,
+			});
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error retrieving Experian token");
+		}
+
+		const twentyFiveMinutesInMillis = 25 * 60 * 1000; 
+		if (token.Item && Date.now() < Number(token.Item.issued_at) + twentyFiveMinutesInMillis) {
```

Comment:
can we break these out so the statement is easier to read like
let tokenPresent = token.Item;
let tokenValidityWindow =  Number(token.Item.issued_at) + twentyFiveMinutesInMillis;

if (tokenIsPresent && Date.now() < tokenValidityWindow) {

Reply:
Does token need to be token?
Is it possible for Dynamo not to return an item

Reply:
I couldn't work out a way for dynamo to just return an item, have addressed the first comment